src/Pulsar.Client/Schema/ProtobufNativeSchema.fs

```diff
@@ -0,0 +1,75 @@
+﻿namespace Pulsar.Client.Schema
+
+
+open System
+open System.IO
+open System.Text
+open System.Text.Json
+open Google.Protobuf.Reflection
+open ProtoBuf
+open Pulsar.Client.Api
+open Pulsar.Client.Common
+
+ type   ProtobufNativeSchemaData(fileDescriptorSet:byte[],
+                                       rootMessageTypeName: string,
+                                       rootFileDescriptorName:string) =      
+    member this.fileDescriptorSet = fileDescriptorSet       
+    member  this.rootMessageTypeName =rootMessageTypeName     
+    member  this.rootFileDescriptorName =rootFileDescriptorName
+
+type internal ProtoBufNativeSchema<'T > () =
+    inherit ISchema<'T>()    
+  
+    let getDescriptor( )=       
+       
+        let gotClassAttribute = Attribute.GetCustomAttributes(typeof<'T>) |>
+                                Array.tryFind
+                                    (fun x -> x :? System.Runtime.Serialization.DataContractAttribute)
+        match gotClassAttribute with
+            Some _ ->
+                let userClassNamespace = typeof<'T>.Namespace
+                let userClassName = typeof<'T>.Name
+                
+                let set = FileDescriptorSet()
+                let protoForType = Serializer.GetProto<'T> () //save it to .proto file nearby
+                let currentPath = Path.GetDirectoryName(typeof<'T>.Assembly.Location)       
+                let protoFileName = userClassName + ".proto"
```

Comment:
Can we avoid writing to the file and write to the memory stream instead?

Reply:
I dont think so. This serialiation is based on protobuf-net protogen. Here, in the start of the program it's explicitly needs input files 
https://github.com/protobuf-net/protobuf-net/blob/14da8b12ed5ca62d84aa16066f74349a9ee7c5d8/src/protogen/Program.cs#L22
and 
https://github.com/protobuf-net/protobuf-net/blob/14da8b12ed5ca62d84aa16066f74349a9ee7c5d8/src/protogen/Program.cs#L117

..next, files are added to work on 

https://github.com/protobuf-net/protobuf-net/blob/14da8b12ed5ca62d84aa16066f74349a9ee7c5d8/src/protogen/Program.cs#L215

I guess if we want to use memory instead files it would need rewriting protobuf-net.Reflection for our needs.